src/api/hooks/categories.ts

```diff
@@ -0,0 +1,14 @@
+import { useQuery } from '@apollo/client'
+import { GET_CATEGORIES } from '@/api/queries'
+import { GetCategories } from '@/api/queries/__generated__/GetCategories'
+
+const useCategories = () => {
+  const { loading, data, error } = useQuery<GetCategories>(GET_CATEGORIES)
```

Comment:
We should probably pass everything from here, not just selected fields. So something like `const { data, ...rest } = ...`